kaggle_environments/envs/llm_20_questions/llm_20_questions.py

```diff
@@ -187,16 +189,24 @@ def interpreter(state, env):
         guessed = False
         if active2.observation.role == GUESSER:
             guessed = guesser_action(active2, inactive2, step)
-            either_guessed = either_guessed or guessed
+            two_guessed = guessed
         else:
             answerer_action(active2, inactive2)
+
         if active2.status in (TIMEOUT, ERROR):
             end_game(active2, inactive2, 0, active2.status, DONE)
         elif end_early:
             end_game(active2, inactive2, 0, DONE, DONE)
         else:
             increment_turn(active2, inactive2, step, guessed)
     
+    # make sure to end the game if either team guessed
+    if one_guessed or two_guessed:
+        if one_guessed and not two_guessed:
+            end_game(active2, inactive2, 0, DONE, DONE)
+        elif two_guessed and not one_guessed:
+            end_game(active1, inactive1, 0, DONE, DONE)
```

Comment:
Shouldn't we also end if both guess?

Reply:
It will already be taken care of in the `guesser_action` where they will be given reward and set status to DONE

Reply:
In that case, you don't need the wrapper `if one_guessed or two_guessed` clause?

Reply:
yeah I can remove that